File: Data Structures & Algorithms/subsets-ii/submission-1.cpp
class Solution {
    vector<vector<int>> ret;
private:
    void backtrack(vector<int>& nums, int start, vector<int>& res){
        if (start == nums.size()) {
            ret.push_back(res);
            return;
        }      

        // include the current num[start]
        res.push_back(nums[start]);
        backtrack(nums, start + 1, res);
        res.pop_back();

        //don't include the current nums[i] (duplicates)
        while(start + 1 < nums.size() && nums[start] == nums[start + 1]){
            start++;
        }backtrack(nums, start + 1, res);
    }
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int> res;
        backtrack(nums, 0, res);
        return ret;
    }
};

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-1.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int ret = 0;
        int l = 0;

        for(int i = 0 ; i < nums.Length ; i++){
            if (nums[i] == 0) k--;
            while(l <= i && k < 0){
                ret = Math.Max(ret, i - l);
                if (nums[l] == 0) k++;
                l++;
            }
        }

        ret = Math.Max(ret, nums.Length - l);
        return ret;
    }
}